app/scripts/app.coffee

```diff
@@ -28,8 +28,8 @@ angular.module('slick', [])
       initialSlide: "@"
       lazyLoad: "@"
       mobileFirst: "@"
-      onBeforeChange: "&"
-      onAfterChange: "&"
+      onBeforeChange: "="
+      onAfterChange: "="
```

Comment:
this is bad. it's adding watchers to the digest cycle for functions that don't change. i wasn't calling setting the function right in the jade file of featuredCarousel (should be `on-after-change="carouselAfterChange()"`. same for beforeChange)

Reply:
With the & binding though, it wasn't recognizing it. I looked into it for a good hour before trying the = binding. Do you have any ideas as to why the & bindings weren't working?

Reply:
it's because of the way i was passing the functions
when using `&` you have to pass it as if you were wanting to call it

Reply:
![](blob:https://carbon.now.sh/a69fde17-f58a-42dc-bc48-1c683c7a49a5)

Reply:
so ```carouselAfterChange()``` instead of ```carouselAfterChange```?